diff --git a/build3/cmake/FindPythonLibs.cmake b/build3/cmake/FindPythonLibs.cmake
index cebac58f42..fcc5724cd0 100644
--- a/build3/cmake/FindPythonLibs.cmake
+++ b/build3/cmake/FindPythonLibs.cmake
@@ -112,7 +112,7 @@ else()
 
   message(STATUS "Looking for versions: ${_PYTHON_VERSIONS}")
 
-  FIND_FILE(_FIND_LIB_PYTHON_PY FindLibPython.py PATHS ${CMAKE_MODULE_PATH} ${CMAKE_ROOT}/Modules)
+  FIND_FILE(_FIND_LIB_PYTHON_PY FindLibPython.py PATHS ${CMAKE_MODULE_PATH} ${CMAKE_ROOT}/Modules ${CMAKE_CURRENT_SOURCE_DIR})
 
   if(NOT _FIND_LIB_PYTHON_PY)
     message(FATAL_ERROR "Could not find required file 'FindLibPython.py'")
diff --git a/CMakeLists.txt b/CMakeLists.txt
index a695b71726..fd9419ce98 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,4 +1,5 @@
-cmake_minimum_required(VERSION 2.4.3)
+cmake_minimum_required(VERSION 3.0.0)
+set(CMAKE_TRY_COMPILE_TARGET_TYPE "STATIC_LIBRARY")
 set(CMAKE_ALLOW_LOOSE_LOOP_CONSTRUCTS true)
 cmake_policy(SET CMP0017 NEW)
 #this line has to appear before 'PROJECT' in order to be able to disable incremental linking
